src/Agent.Worker/TestResults/NunitResultReader.cs

```diff
@@ -267,10 +243,94 @@ private List<TestCaseResultData> FindTestCaseNodes(XmlNode startNode, string hos
             return results;
         }
 
-        public bool AddResultsFileToRunLevelAttachments
+        public string InnerTestSuiteNodeName { get; set; }
+        public string TestCaseNodeName { get; set; }
+        public string RootNodeName { get; set; }
+
+    }
+
+
+    public interface INUnitResultsXmlReader
+    {
+        TestRunData GetTestRunData(string filePath, XmlDocument doc, XmlNode testResultsNode, TestRunContext runContext, bool addResultsAsAttachments);
+    }
+
+    internal class NUnit2ResultsXmlReader: NUnitResultsXmlReader, INUnitResultsXmlReader
+    {
+        public new TestRunData GetTestRunData(string filePath, XmlDocument doc, XmlNode testResultsNode, TestRunContext runContext, bool addResultsAsAttachments)
+        {
+            this.InnerTestSuiteNodeName = "results/test-suite";
+            this.TestCaseNodeName = "results/test-case";
+            this.RootNodeName = "test-results";
+
+            return base.GetTestRunData(filePath, doc, testResultsNode, runContext, addResultsAsAttachments);
+        }
+    }
+
+    internal class NUnit3ResultsXmlReader: NUnitResultsXmlReader, INUnitResultsXmlReader
+    {
+        public new TestRunData GetTestRunData(string filePath, XmlDocument doc, XmlNode testResultsNode, TestRunContext runContext, bool addResultsAsAttachments)
         {
-            get;
-            set;
+            this.InnerTestSuiteNodeName = "./test-suite";
+            this.TestCaseNodeName = "./test-case";
+            this.RootNodeName = "test-run";
+
+            return base.GetTestRunData(filePath, doc, testResultsNode, runContext, addResultsAsAttachments);
```

Comment:
@CharliePoole that was my concern although I haven't had time to look right through this PR. The unit tests for this only cover simple pass/fails. Correct me if I am wrong, but the syntax for the various types of errors and skipped tests have changed, so I expect that this code will work for simple pass/fail, but get errors, ignored tests, explicit, etc. incorrect.